File: TemplateNetCore-main/WebAdmin/Modelos/Cotizacion.vb
Namespace MercanciaSegura.DOM.Modelos
    Public Class Cotizacion
        Public Property CotizacionId As Integer
        Public Property PolizaId As Integer
        Public Property FechaCotizacion As DateTime
        Public Property ClienteId As Integer
        Public Property nombreCliente As String
        Public Property BeneficiarioPreferenteId As Integer?
        Public Property MonedaId As Integer

        Public Property VigenciaDel As DateTime?
        Public Property VigenciaHasta As DateTime?
        Public Property FechaCancelacion As DateTime?
        Public Property PrimaServicioDeAseguramiento As Decimal?
        Public Property Subtotal As Decimal?
        Public Property IVA As Decimal?
        Public Property Total As Decimal?
        Public Property GastosExpedicion As Decimal?
        Public Property CotizacionMercancia As CotizacionMercancia
        Public Property CotizacionContenedor As List(Of CotizacionContenedor)


    End Class

End Namespace

File: TemplateNetCore-main/MercanciaSegura.DOM/Modelos/Cotizacion/Cotizacion.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MercanciaSegura.DOM.Modelos.Cotizacion
{
    [Table("Cotizacion")]
    public class Cotizacion
    {
        [Key]
        [Column("Cotizacion_ID")]
        public int CotizacionId { get; set; }

        [Column("Poliza_ID")]
        public int? PolizaId { get; set; }

        [ForeignKey(nameof(PolizaId))]
        public Poliza.Poliza? Poliza { get; set; }

        [Column("Fecha_Cotizacion")]
        public DateTime FechaCotizacion { get; set; }

        [Column("Cliente_ID")]
        public int? ClienteId { get; set; }

        [ForeignKey(nameof(ClienteId))]
        public Cliente.Cliente? Cliente { get; set; }

        [Column("Beneficiario_Preferente_ID")]
        public int? BeneficiarioPreferenteId { get; set; }

        [ForeignKey(nameof(BeneficiarioPreferenteId))]
        public Cliente.BeneficiarioPreferente? BeneficiarioPreferente { get; set; }

        [Column("Moneda_ID")]
        public int? MonedaId { get; set; }

        [ForeignKey(nameof(MonedaId))]
        public Poliza.Moneda? Moneda { get; set; }

        [Column("Vigencia_Del")]
        public DateTime? VigenciaDel { get; set; }

        [Column("Vigencia_Hasta")]
        public DateTime? VigenciaHasta { get; set; }

        [Column("Fecha_Cancelacion")]
        public DateTime? FechaCancelacion { get; set; }

        [Column("Prima_Servicio_De_Aseguramiento", TypeName = "decimal(18,4)")]
        public decimal? PrimaServicioDeAseguramiento { get; set; }

        [Column("Subtotal", TypeName = "decimal(18,4)")]
        public decimal? Subtotal { get; set; }

        [Column("IVA", TypeName = "decimal(18,4)")]
        public decimal? IVA { get; set; }

        [Column("Total", TypeName = "decimal(18,4)")]
        public decimal? Total { get; set; }

        [Column("Gastos_Expedicion", TypeName = "decimal(18,2)")]
        public decimal? GastosExpedicion { get; set; }
        public CotizacionMercancia? CotizacionMercancia { get; set; }
        public List<CotizacionContenedor>? CotizacionContenedor { get; set; }
    }
}